docs/content/concepts/assets/asset-checks.mdx

```diff
@@ -0,0 +1,141 @@
+---
+title: Asset checks | Dagster
+description: Asset checks are a way to define expectations about the quality of your data.
+---
+
+# Asset checks (Experimental)
+
+Dagster allows you to define and execute data quality checks on your [software-defined assets](/concepts/assets/software-defined-assets). Each asset check verifies some property of a data asset, e.g. that is has no null values in a particular column.
+
+When viewing an asset in Dagster’s UI, you can see all of its checks, and whether they’ve passed, failed, or haven’t run.
+
+## Defining asset checks
+
+### Asset check that executes in its own op
+
+These are the bare APIs for defining an asset and an asset check that executes in its own op.
```

Comment:
I find this a tiny bit confusing, because what follows is a code snippet, not a list of APIs. I would lean towards just jumping in to the "realistic" example, minus the metadata to keep it small, because it's still pretty small and the API is easier to understand how to use in context.

Reply:
I think removing the metadata would be a bummer bc it shows a perfect use...

"This is the syntax for defining an asset check"? Personally I like showing a skeleton then an example, because I think even without metadata it's too large of a code block to get a quick glimpse of the apis in action. Not strongly held

Reply:
My rough framework for these is that each code snippet should:
- Get across a unique nugget of understanding.
- With its associated descriptive paragraph, be self-contained enough to get that across.

In this case, I think the goal of the first snippet should be to get across a few interrelated facts:
- There's an `@asset_check` decorator, which defines an asset check that references a particular asset.
- The `@asset_check`-decorated function includes the code that implements the asset check.
- This code is expected to verify some property of the asset and return success based on that property.

I think the point of including the Pandas stuff is to get across these facts. If we can find a way to get them across that's less verbose, then leaving out the Pandas stuff would be good too.

One of my concerns with what's currently here is that the first snippet and its associated description might not include enough information on its own for the user to understand what's going on.

> I think removing the metadata would be a bummer bc it shows a perfect use...

We could include a separate snippet afterwards with the topic "Adding metadata to a check result" or similar?